tests/tests/test_demo_artifact.py

```diff
@@ -148,5 +149,5 @@ def demo_artifact_installation(self, run_demo_script):
     def demo_up_down_up(self, run_demo_script):
         """Test that bringing the demo environment up, then down, then up succeeds"""
         self.demo_artifact_upload(run_demo_script)
-        self.demo_artifact_upload(run_demo_script)
+        self.demo_artifact_upload(run_demo_script, exit_cond="The user already exists")
```

Comment:
To me, this seems to change the purpose of the test, and I don't see this mentioned in the commit message.

The test documentation does not help to figure out what are we actually verifying here: "Test that bringing the demo environment up, then down, then up succeeds".

So what is this test trying to do? Is the purpose of the test to see that a rerun of the demo script no user is created but the containers are anyway up and running?

Reply:
Hmm, ur right. It is a poor name. I will split it up into two commits, and rename the test.